workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/HandleCreatePlayerRequestSystem.cs

```diff
@@ -12,83 +12,86 @@ namespace Improbable.Gdk.PlayerLifecycle
             public PlayerCreator.CreatePlayer.ReceivedRequest createPlayerRequest;
         }
 
         protected override void OnUpdate()
         {
-            for (var i = 0; i < createPlayerData.Length; i++)
+            HandlePlayerCreateRequests();
+            HandlePlayerCreateEntityResponses();
+        }
+
+        private void HandlePlayerCreateRequests()
+        {
+            if (PlayerLifecycleConfig.CreatePlayerEntityTemplate == null)
             {
-                var requests = createPlayerData.CreatePlayerRequests[i].Requests;
-                var createEntitySender = createPlayerData.CreateEntitySender[i];
+                throw new InvalidOperationException(Errors.PlayerEntityTemplateNotFound);
+            }
 
-                foreach (var request in requests)
-                {
-                    if (PlayerLifecycleConfig.CreatePlayerEntityTemplate == null)
+            var requests = commandSystem.GetRequests<PlayerCreator.CreatePlayer.ReceivedRequest>();
+            foreach (var request in requests)
+            {
+                var playerEntityTemplate = PlayerLifecycleConfig.CreatePlayerEntityTemplate(
+                    request.CallerWorkerId,
+                    request.Payload.Position
+                );
+
+                var entityRequest = new WorldCommands.CreateEntity.Request
+                (
+                    playerEntityTemplate,
+                    context: new PlayerCreationRequestContext
                     {
-                        throw new InvalidOperationException(Errors.PlayerEntityTemplateNotFound);
+                        createPlayerRequest = request
                     }
+                );
 
-                    var playerEntity = PlayerLifecycleConfig.CreatePlayerEntityTemplate(request.CallerWorkerId,
-                        request.Payload.Position);
-                    createEntitySender.RequestsToSend.Add(new WorldCommands.CreateEntity.Request
-                    (
-                        playerEntity,
-                        context: new PlayerCreationRequestContext
-                        {
-                            createPlayerRequest = request
-                        }
-                    ));
-                }
+                commandSystem.SendCommand(entityRequest);
             }
+        }
 
-            for (var i = 0; i < entityCreationResponseData.Length; ++i)
+        private void HandlePlayerCreateEntityResponses()
+        {
+            var responses = commandSystem.GetResponses<WorldCommands.CreateEntity.ReceivedResponse>();
+            foreach (var response in responses)
             {
-                var entityCreationResponses = entityCreationResponseData.CreateEntityResponses[i];
-                var responder = entityCreationResponseData.CreatePlayerResponders[i];
-
-                foreach (var receivedResponse in entityCreationResponses.Responses)
+                if (!(response.Context is PlayerCreationRequestContext requestContext))
```

Comment:
This raises an interesting idea on whether we should somehow "type" world commands.